stdnum/li/peid.py

```diff
@@ -0,0 +1,87 @@
+# peid.py - library for Liechtenstein Personenidentifikationsnummer
+#
+# Copyright (C) 2020 Matthias Schmid
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301 USA
+
+"""PEID (Liechtenstein tax code for individuals and entities).
+
+The Personenidentifikationsnummer (PEID) is an numeric code up to 12
+digits used to identify entities and individuals residing in
+Liechtenstein.
+
+The first digits might be omitted if they correspond to the numeral
+zero. Nevertheless the minimum number of digits is four.
+
+
+More information:
+
+* http://www.oecd.org/tax/automatic-exchange/crs-implementation-and-assistance/tax-identification-numbers/
+* https://www.oecd.org/tax/automatic-exchange/crs-implementation-and-assistance/tax-identification-numbers/LIECHTENSTEIN%20TIN.pdf
+
+>>> compact('00001234567')
+'1234567'
+
+>>> validate('1234567')  # personal number or entity number
+'1234567'
+>>> validate('00001234567')
+'1234567'
+
+>>> validate('00001234568913454545')
+Traceback (most recent call last):
+    ...
+InvalidLength: The number has an invalid length.
+
+>>> validate('568ABC91345')
+Traceback (most recent call last):
+    ...
+InvalidFormat: The number has an invalid format.
+"""
+
+import re
+
+from stdnum.exceptions import *
+from stdnum.util import clean
+
+
+# regular expression for matching Liechtenstein Personenidentifikationsnummer
+_code_re = re.compile(r'^[0-9]{4,12}$')
+
+
+def compact(number):
+    """Convert the number to the minimal representation. This strips the
+    number of any valid separators and removes surrounding whitespace."""
+    return clean(number).strip().lstrip('0').replace('.', '', 2)
```

Comment:
You can pass `clean` the characters you want to get removed, like `clean(number, ' .')` which will remove both the whitespace and the period characters, so the `strip` and the `replace` are unnecessary. Note that the `lstrip` is still required.

Reply:
On a second thought maybe keeping `strip` is not a bad idea since it would remove newlines and such.

Reply:
I have been using `.replace('.', '', 2)` instead of `clean(number, ' .')` because I wanted to make sure only 2 dots are being removed. as if 3 or more would be there it would be invalid according to the samples of #125 . as far as I know according to the spec of the PEID there is actually no "dot" in at all.

Reply:
I don't see any problem with removing all the dots.

Reply:
fixed compact according to @unho 's suggestions